dlt/extract/schema.py

```diff
@@ -220,12 +224,19 @@ def new_table_template(
             columns = ensure_table_schema_columns_hint(columns)
             if not callable(columns):
                 columns = columns.values()  # type: ignore
+            is_table_complete = len([c for c in columns if c.get("name") and c.get("data_type")]) # type: ignore
         else:
             validator = None
+            is_table_complete = False
+
+        # freeze the columns if we have a fully defined table and no other explicit contract
+        if not schema_contract and is_table_complete:
```

Comment:
this is how I enforce the pydantic models (and other schema column settings) automatically. I am not quite sure if this covers all cases really and if the "is_table_complete" does the right thing.

Reply:
also i allow the table to be created if it adheres to the schema.. I think this makes more sense